public/app/features/datasources/components/DataSourceTestingStatus.tsx

```diff
@@ -97,13 +101,16 @@ export function DataSourceTestingStatus({ testingStatus, exploreUrl, dataSource
     });
   };
 
+  const instanceState = useDataSourceInstance(dataSource.uid);
```

Comment:
Ok so this is the part where I'd love to get the most feedback on – this hook essentially calls `useAsync()` to fetch the _instance_ of a data source so we get call the `renderHealthCheckDetails()` function.

This cause the test message to re-render while we're off to fetch that instance – it's imperceptible in any real-world scenario but complicates the testing of the component (and parent components) to a degree since I have to wrap the assertions we make with `await waitFor(() => { })` 

I'd appreciate any thoughts or comments here.

Reply:
We wouldn't need this if we would go with the "register a component approach". Then you can just get the component via `plugin.components.DataSourceTestingDetails`.